model/src/order.rs

```diff
@@ -206,9 +206,8 @@ pub struct OrderCreation {
     #[serde(with = "u256_decimal")]
     pub buy_amount: U256,
     pub valid_to: u32,
-    #[derivative(Debug(format_with = "debug_app_data"))]
```

Comment:
Because we no longer need `Derivative` here anymore, I would get rid of:
```rust
#[derive(Eq, PartialEq, Clone, Derivative, Deserialize, Serialize, Hash)]
#[derivative(Debug)]
```

in favour of
```rust
#[derive(Eq, PartialEq, Clone, Debug, Deserialize, Serialize, Hash)]
```

Previously, `Derivative` was used to override the `Debug` implementation to format the `app_data` field as hexadecimal, but this is no longer needed as the `AppId` newtype implements `Debug` as formatted hexadecimal anyway.